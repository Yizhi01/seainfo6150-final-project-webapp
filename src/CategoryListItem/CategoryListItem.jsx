import React from "react";
import { Link } from "react-router-dom";
import styles from "./CategoryListItem.module.css";
import PropTypes from "prop-types";

const CategoryListItem = (props) => {
  const { id, name, url } = props; 
  return (
  <div>
    <Link to={"/category/" + id +'/' + name} className={styles.categoryItem}>
      <img
        className={styles.categoryImage}
        src={url}
        alt={name}
      />
      <p className={styles.categoryName}>{name}</p>
    </Link>
  </div>);
};

CategoryListItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default CategoryListItem;
