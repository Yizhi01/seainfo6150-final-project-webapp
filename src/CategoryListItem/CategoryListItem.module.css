.categoryItem {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  text-decoration: none;
  color: #222831;
}

.categoryImage {
  width: 80%;
  border-radius: 100%;
}

.categoryName {
  margin: 20px 0;
  font-family: 'Castoro', serif;
  font-size: 1.2em;
  text-transform: capitalize;
}

.categoryItem:hover {
  opacity: 0.75;
  color: teal;
}