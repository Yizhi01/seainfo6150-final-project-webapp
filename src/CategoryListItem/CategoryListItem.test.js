import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CategoryListItem from "./CategoryListItem.jsx";

it("renders the CategoryListItem component correctly", () => {
  const { container } = render(
    <Router>
      <CategoryListItem id="1" name="breakfast" url="https://source.unsplash.com/Wns3U-oVoLA/960x960" />
    </Router>
   
  );
  expect(container).toMatchSnapshot();
});