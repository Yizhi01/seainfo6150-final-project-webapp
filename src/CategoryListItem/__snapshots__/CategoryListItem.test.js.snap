// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CategoryListItem component correctly 1`] = `
<div>
  <div>
    <a
      class="categoryItem"
      href="/category/1/breakfast"
    >
      <img
        alt="breakfast"
        class="categoryImage"
        src="https://source.unsplash.com/Wns3U-oVoLA/960x960"
      />
      <p
        class="categoryName"
      >
        breakfast
      </p>
    </a>
  </div>
</div>
`;
